components/ILIAS/Contact/classes/class.ilMailSearchObjectGUI.php

```diff
@@ -136,137 +137,48 @@ protected function addPermission(array $a_obj_ids): void
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-                );
-            }
-
-            if ($obj_ids !== []) {
-                $this->mailObjects();
-            } else {
-                $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_' . $this->getObjectType()));
-                $this->showMyObjects();
-            }
-        } elseif ($view === $this->getObjectType() . '_members') {
-            $usr_ids = [];
-            if ($this->http->wrapper()->query()->has('search_members')) {
-                $usr_ids = [
-                    $this->http->wrapper()->query()->retrieve(
-                        'search_members',
-                        $this->refinery->kindlyTo()->int()
-                    )
-                ];
-            } elseif ($this->http->wrapper()->post()->has('search_members')) {
-                $usr_ids = $this->http->wrapper()->post()->retrieve(
-                    'search_members',
-                    $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-                );
-            }
-
-            if ($usr_ids !== []) {
-                $this->mailMembers();
-            } else {
-                $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_one_entry'));
-                $this->showMembers();
-            }
+        $usr_ids = $this->http->wrapper()->query()->has('contact_mailinglist_search_members_ids')
+            ? $this->http->wrapper()->query()->retrieve(
+                'contact_mailinglist_search_members_ids',
+                $this->refinery->kindlyTo()->dictOf($this->refinery->kindlyTo()->int())
+            ) : [];
+
+        if ($usr_ids !== []) {
+            $this->addPermission(array_unique($usr_ids));
         } else {
-            $this->showMyObjects();
+            $this->tpl->setOnScreenMessage('info', $this->lng->txt('mail_select_one_entry'));
+            $this->showMembers();
         }
     }
 
-    protected function mailObjects(): void
+    private function mailObjects(): void
     {
         $members = [];
         $mail_data = $this->umail->retrieveFromStage();
 
-        $obj_ids = [];
-        if ($this->http->wrapper()->query()->has('search_' . $this->getObjectType())) {
-            $obj_ids = [
-                $this->http->wrapper()->query()->retrieve(
-                    'search_' . $this->getObjectType(),
-                    $this->refinery->kindlyTo()->int()
-                )
-            ];
-        } elseif ($this->http->wrapper()->post()->has('search_' . $this->getObjectType())) {
-            $obj_ids = $this->http->wrapper()->post()->retrieve(
-                'search_' . $this->getObjectType(),
-                $this->refinery->kindlyTo()->listOf($this->refinery->kindlyTo()->int())
-            );
-        }
+        $obj_ids = $this->http->wrapper()->query()->has('contact_mailinglist_search_obj_ids')
```

Comment:
This is not a `dictOf`, but a `listOf`. Please also try to use `byTring` and `always` to avoid the `has` check.